d2/src/main/java/com/linkedin/d2/balancer/util/TogglingLoadBalancer.java

```diff
@@ -54,7 +54,7 @@
 
 public class TogglingLoadBalancer implements LoadBalancer, HashRingProvider, ClientFactoryProvider, PartitionInfoProvider, WarmUpService, ClusterInfoProvider
 {
-  private final LoadBalancer _balancer;
+  private final SimpleLoadBalancer _balancer;
```

Comment:
Why do we need to introduce a strong coupling between TogglingLoadBalancer vs SimpleLoadBalancer?

Reply:
Because the fact is that we are always passing a SimpleLoadBalancer. This is for dev's benefit to understand how load balancers are wrapped in layers. Otherwise it's so hard to tell which load balancer uses which one, we have so many layers wrapped one by one. When I see this togglingLoadBalancer only accepts SimpleLoadBalancer in its constructor, I just want to make it more explicit.